src/main/java/seedu/address/logic/commands/SearchPatientCommand.java

```diff
@@ -0,0 +1,187 @@
+                Person personFound = personList.get(0);
+                boolean isPersonFound = false;
+                for (Person person : personList) {
+                    String personName = person.getName().toString().trim().toLowerCase();
+                    if (personName.equals(nameToSearch)) {
+                        personFound = person;
+                        isPersonFound = true;
+
+                    }
+                }
+                if (!isPersonFound) {
+                    throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+                } else {
+                    return new CommandResult(String.format(MESSAGE_SEARCH_PERSON_SUCCESS, personFound));
+                }
+
+
+            case (AREA_IS_TEMPERATURE):
+                double startingTemperature = searchPersonDescriptor.getTemperatureRange().getStartingTemperature();
+                double endingTemperature = searchPersonDescriptor.getTemperatureRange().getEndingTemperature();
+                for (Person person : personList) {
+                    if (person.getTemperature().getValue() >= startingTemperature &&
+                            person.getTemperature().getValue() <= endingTemperature ) {
+                        personWithinTemperatureRange.add(person);
+                    }
+                }
+
+                if (personWithinTemperatureRange.isEmpty()) {
+                    throw new CommandException(MESSAGE_PERSON_NOT_FOUND);
+                }
+                else {
+                    return new CommandResult(MESSAGE_SEARCH_PERSON_LIST_SUCCESS +
+                            getListOutput(personWithinTemperatureRange));
+                }
+
+            case (TOO_MANY_AREA):
+                return new CommandResult(Messages.MESSAGE_TOO_MANY_COMMANDS);
+
+
+        }
+
+        return null;
+    }
+
+    public String getListOutput(ArrayList<Person> list) {
+        StringBuilder outputString = new StringBuilder();
+        for (int i = 0 ; i < list.size(); i++) {
+            outputString.append(String.format("%d.%s\n", i + 1, list.get(i)));
+        }
+        return outputString.toString();
+    }
+
+    public int ConfirmArea(SearchPersonDescriptor searchPersonDescriptor) {
+        if (searchPersonDescriptor.getOptionalName().isEmpty() && searchPersonDescriptor.getOptionalTemperatureRange().isEmpty()) {
+            return AREA_NOT_FOUND;
+        }
+        else if (searchPersonDescriptor.getOptionalName().isPresent() && searchPersonDescriptor.getOptionalTemperatureRange().isEmpty()) {
+            return AREA_IS_NAME;
+        }
+        else if (searchPersonDescriptor.getOptionalName().isEmpty() && searchPersonDescriptor.getOptionalTemperatureRange().isPresent()) {
+            return AREA_IS_TEMPERATURE;
+        }
+        else {
+            return TOO_MANY_AREA;
+        }
+
+    }
+
+
+    public static class SearchPersonDescriptor {
```

Comment:
maybe you want to change the name to SearchPatientDescriptor? can rightclick, ``refactor`` -> ``rename``

Reply:
yup! SearchPatientDescriptor is a better name